director/redirect.go

```diff
@@ -278,21 +280,64 @@ func RedirectToOrigin(ginCtx *gin.Context) {
 
 }
 
+func checkHostnameRedirects(c *gin.Context, incomingHost string) {
+	oRedirectHosts := param.Director_OriginResponseHostnames.GetStringSlice()
+	cRedirectHosts := param.Director_CacheResponseHostnames.GetStringSlice()
+
+	for _, hostname := range oRedirectHosts {
+		if hostname == incomingHost {
+			if !strings.HasPrefix(c.Request.URL.Path, "/api/v1.0/director/") {
+				c.Request.URL.Path = "/api/v1.0/director/origin" + c.Request.URL.Path
+				RedirectToOrigin(c)
+				c.Abort()
+				log.Debugln("Director is serving an origin based on incoming 'Host' header value of '" + hostname + "'")
+				return
+			}
+		}
+	}
+	for _, hostname := range cRedirectHosts {
+		if hostname == incomingHost {
+			if !strings.HasPrefix(c.Request.URL.Path, "/api/v1.0/director/") {
+				c.Request.URL.Path = "/api/v1.0/director/object" + c.Request.URL.Path
+				RedirectToCache(c)
+				c.Abort()
+				log.Debugln("Director is serving a cache based on incoming 'Host' header value of '" + hostname + "'")
+				return
+			}
+		}
+	}
+}
+
 // Middleware sends GET /foo/bar to the RedirectToCache function, as if the
 // original request had been made to /api/v1.0/director/object/foo/bar
 func ShortcutMiddleware(defaultResponse string) gin.HandlerFunc {
 	return func(c *gin.Context) {
 		// If this is a request for getting public key, don't modify the path
 		// If this is a request to the Prometheus API, don't modify the path
-		if strings.HasPrefix(c.Request.URL.Path, "/.well-known") ||
-			strings.HasPrefix(c.Request.URL.Path, "/api/v1.0/prometheus") {
+		if strings.HasPrefix(c.Request.URL.Path, "/.well-known/") ||
+			(strings.HasPrefix(c.Request.URL.Path, "/api/v1.0/") && !strings.HasPrefix(c.Request.URL.Path, "/api/v1.0/director/")) {
 			c.Next()
 			return
 		}
+
+		if param.Director_HostAwareRedirects.GetBool() {
```

Comment:
Why is this an option?  What's the harm in unconditionally enabling this?

In other words, any time an admin wants to enable this, they have to set two parameters (flip a boolean then add a name to a list).  It seems that can be simplified to one action -- adding a name to the list implies it should be used, no?